main.cpp

```diff
@@ -142,6 +142,12 @@ int main(int argc, char *argv[]) {
 
 	auto statistics = optWriter.GetStatistics();
 
+	if (statistics.bytesTotal_ != volumeSize) {
```

Comment:
It's allowed not to give `volumeSize`, so this should also test for `volumeSize != 0`, right?

Reply:
It's only allowed when not reading from a pipe, right? Otherwise the code doesn't know when to stop reading because `stat()` on the pipe descriptor doesn't tell size.

Reply:
In practice it's always going to be reading from a pipe, because that's how the Update Module protocol delivers the data. The pipe will get EOF once it has reached the end. So we simply need to disable checking of the size if `volumeSize` wasn't given.

Reply:
Added now.